### ISSUE TITLE
Add proper type definitions based on ISA spec

### ISSUE THREAD

**Issue by knejadfard** [knejadfard/rkern#35]
Initially I had encountered an issue using types such as `uint8_t`. It looks like these types are defined in `<cstdint>` and should be available in freestanding mode. If they are not, then perhaps this is a sign for an issue with my development setup.

**Comment by knejadfard** [comment]
Got GCC to work fine. Clang with libc++ is not working at the moment.
It looks like the issue has to do with the way I have built libc++ in my local environment. The error I get is:
```
[33%] Building CXX object CMakeFiles/rkern.dir/src/kernel.cpp.o
In file included from /home/kian/workspace/rkern/src/kernel.cpp:6:
In file included from /home/kian/workspace/rkern/include/vga.hpp:4:
In file included from /home/kian/binaries/llvm-10.0.1/bin/../include/c++/v1/cstdint:143:
/home/kian/binaries/llvm-10.0.1/bin/../include/c++/v1/__config:1132:6: error: "No thread API"
#    error "No thread API"
     ^
1 error generated.
make[2]: *** [CMakeFiles/rkern.dir/build.make:82: CMakeFiles/rkern.dir/src/kernel.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:95: CMakeFiles/rkern.dir/all] Error 2
make: *** [Makefile:103: all] Error 2
```
It looks like, if I build libc++ with `-DLIBCXX_ENABLE_THREADS=OFF`, the macro conditional statements in `c++/v1/__config` would not result in printing the above error message. Essentially, I need to build libc++ with threads disabled.

**Comment by knejadfard** [comment]
It is a better idea to define data types according to official RISC-V conventions/standard that can be found at https://www2.eecs.berkeley.edu/Pubs/TechRpts/2014/EECS-2014-54.pdf.
Located at section 18.1.
